src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -455,32 +455,35 @@ public void KeepAlive_SetThenGetBoolean_ValuesMatch(Uri remoteServer)
         [InlineData(null)]
         [InlineData(false)]
         [InlineData(true)]
-        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "dotnet/corefx #19225")]
-        public void KeepAlive_CorrectConnectionHeaderSent(bool? keepAlive)
+        public async Task KeepAlive_CorrectConnectionHeaderSent(bool? keepAlive)
         {
-            HttpWebRequest request = WebRequest.CreateHttp(Test.Common.Configuration.Http.RemoteEchoServer);
-
-            if (keepAlive.HasValue)
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
             {
-                request.KeepAlive = keepAlive.Value;
-            }
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+                request.Proxy = null; // Don't use a proxy since it might interfere with the Connection: headers.
+                if (keepAlive.HasValue)
+                {
+                    request.KeepAlive = keepAlive.Value;
+                }
 
-            using (var response = (HttpWebResponse)request.GetResponse())
-            using (var body = new StreamReader(response.GetResponseStream()))
-            {
-                string content = body.ReadToEnd();
+                Task<WebResponse> getResponseTask = request.GetResponseAsync();
+                Task<List<string>> serverTask = server.AcceptConnectionSendResponseAndCloseAsync();
+
+                await TaskTimeoutExtensions.WhenAllOrAnyFailed(new Task[] { getResponseTask, serverTask });
+
+                List<string> requestLines = await serverTask;
                 if (!keepAlive.HasValue || keepAlive.Value)
                 {
                     // Validate that the request doesn't contain Connection: "close", but we can't validate
```

Comment:
Nit: should we change the comment to `...doesn't contain "Connection: Close"`?

Reply:
Fixed.  Updated comment.